### PR TITLE
#22  As a host, I want to be able to search among energisers by name / description Create a new Searchfilter-JustinasBrazauskas

### LINKED ISSUE #22: As a host, I want to be able to search among  energisers by name / description

### PR DESCRIPTION
**Pull request by JustinasBrazauskas**
## Naming Rules

Name your PR like this: ISSUENUMBER-TITLE-YOURNAME

## Description

Create a new Searchfilter

## Related to

Make sure you include the issue number with a hash sign # so Github can link this PR to the right issue, like this:

Fixes #22

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have carefully reviewed my own code
- [ ] I have commented my code
- [ ] I have updated any documentation

### CODE REVIEW

File: client/src/pages/Energisers.js [hunk 1]
```diff
@@ -18,14 +19,16 @@ function Energisers() {
 			})
 			.then((body) => {
 				setAllEnergizers(body);
+				setInitialAllEnergizers(body);
```
Comment (bertilnilsson) [inline client/src/pages/Energisers.js]: Why is this data being stored in two different states that are duplicates?

File: client/src/pages/Home.js [hunk 2]
```diff
@@ -8,12 +8,16 @@ import "./Home.css";
...
 	const [ratedEnergizers, setRatedEnergizers] = useState([]);
+	const [initialRecentEnergizers, setInitialRecentEnergizers] = useState([]);
+	const [initialRatedEnergizers, setInitialRatedEnergizers] = useState([]);
 	useEffect(() => {
 		fetch("/api/energizers?sort_by=recent")
 			.then((response) => response.json())
 			.then((energizers) => {
 				const mostRecentEnergizers = energizers.slice(0, 2);
 				setMostRecentEnergizers(mostRecentEnergizers);
+				setInitialRecentEnergizers(mostRecentEnergizers);
```
Comment (bertilnilsson) [inline client/src/pages/Home.js]: Same question - why is this data duplicated into two different states? It seems the only reason is that one state is being trimmed but that could easily happen when the state is being used.

**Review by bertilnilsson — COMMENTED** [review]
It seems that the structure of this in terms of the use of states is a bit backwards. Probably what makes more sense is to have only the query state of the Navbar update when someone is doing a search and then reference that query field to filter the result on each page.. rather than passing multiple different states into the Navbar and having it filter that data.

File: client/src/pages/Home.js (hunk 2 above)
Comment (JustinasBrazauskas) [reply client/src/pages/Home.js]: I've been updated query, so should be more clearly now , it's working!

**Pull request merged**